cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java

```diff
@@ -115,6 +117,21 @@ public ProcessResult run(boolean capture) {
     }
     // pragmatic solution to avoid copying lists/arrays
     this.arguments.add(0, this.executable.toString());
+
+    String executableName = this.executable.toString();
+    String fileExtension = FilenameUtil.getExtension(this.executable.toString());
+    boolean isBashScript = "sh".equals(fileExtension) || hasSheBang(executableName);
+    if(isBashScript && this.context.getSystemInfo().isWindows()) {
+      String bashPath = findPathFor("GitForWindows");
+      if (bashPath == null) {
+        bashPath = findPathFor("Cygwin\\setup");
+      }
+      if (bashPath == null) {
+        throw new IllegalStateException("Could not find bash. Please install git for Windows and rerun.");
+      }
+      this.arguments.add(0, bashPath);
```

Comment:
You need to also add `-c` option (`bash.exe -c my-script.sh myarg1 myarg2`).
Try this in your git-bash:
```
$ bash.exe -c "echo hi"
hi
```

Reply:
As the entire "command" has to be one after argument to `bash -c`, I guess that you have to do something like this:
```
this.arguments.add(0, this.executable.toString());
List<String> args = this.arguments;
if (isBashScript) {
  // new logic goes here
  args = new ArrayList();
  String bash = "bash";
  if (this.context.getSystemInfo().isWindows()) {
    // ... your code to find bash on windows goes here ...
    bash = findBashOnWindows();
  }
  args.add(bash);
  args.add("-c");
  args.add(this.arguments.join(" "));
}
this.processBuilder.command(args);
```

Reply:
Apparently in ProcessBuilder `-c` is only used when running a bash command, and not a bash script. 
`processBuilder.command("bash", "-c", "ls")` works, but `processBuilder.command("bash", "-c", pathToScript)` returns command not found.
With that said, I went back to my old variation of using `processBuilder.command(pathToBash, pathToScript, args)` which works perfectly.